docs: improve install script docs (#1136)
prefix-dev/pixi: docs/index.md

If you have any issues building because of the dependency on `rattler` checkout
its [compile steps](https://github.com/mamba-org/rattler/tree/main#give-it-a-try).

### Installer script options

=== "Linux & macOS"

    The installation script has several options that can be manipulated through environment variables.

    | Variable             | Description                                                                        | Default Value         |
    |----------------------|------------------------------------------------------------------------------------|-----------------------|

>     | `VERSION`            | The version of pixi getting installed, can be used to up- or down-grade.           | `latest`              |
>     | `PIXI_HOME`          | The location of the binary folder.                                                  | `$HOME/.pixi`         |
>     | `PIXI_ARCH`          | The architecture the pixi version was built for.                                    | `uname -m`            |
>     | `PIXI_NO_PATH_UPDATE`| If set the `$PATH` will not be updated to add `pixi` to it.                         |                       |
>     | `TMP_DIR`            | The temporary directory the script uses to download to and unpack the binary from.   | `/tmp`                |

Reviewer: not that it matters 🤔 

![image](https://github.com/prefix-dev/pixi/assets/29506042/eb695e83-bbe9-4527-86db-aef470fe739c)
```suggestion
    | `VERSION`            | The version of pixi getting installed, can be used to up- or down-grade.           | `latest`              |
    | `PIXI_HOME`          | The location of the binary folder.                                                 | `$HOME/.pixi`         |
    | `PIXI_ARCH`          | The architecture the pixi version was built for.                                   | `uname -m`            |
    | `PIXI_NO_PATH_UPDATE`| If set the `$PATH` will not be updated to add `pixi` to it.                        |                       |
    | `TMP_DIR`            | The temporary directory the script uses to download to and unpack the binary from. | `/tmp`                |
```

    |------------------|----------------------|-----------------------------------------------------------------------------------|-----------------------------|
    | `PixiVersion`    | `PIXI_VERSION`       |The version of pixi getting installed, can be used to up- or down-grade.         | `latest`                    |
    | `PixiHome`       | `PIXI_HOME`          | The location of the installation.                                                 | `$Env:USERPROFILE\.pixi`   |
    | `NoPathUpdate`   |                      | If set, the `$PATH` will not be updated to add `pixi` to it.                      |                             |

    For example, set the version using:

    ```powershell
    iwr -useb https://pixi.sh/install.ps1 | iex -Args "-PixiVersion v0.18.0"
    ```
## Update

Updating is as simple as installing, rerunning the installation script gets you the latest version.

```shell
pixi self-update
```
Or get a specific pixi version using:
```shell
pixi self-update --version x.y.z
```

!!! note
    If you used a package manager like `brew`, `mamba`, `conda`, `paru` to install `pixi`.

>     Then use their builtin update mechanism. e.g. `brew upgrade pixi`.

Reviewer 2: ```suggestion
    It's preferable to use the built-in update mechanism. e.g. `brew upgrade pixi`.
```

    | `PIXI_ARCH`          | The architecture the pixi version was built for.                                   | `uname -m`            |
    | `PIXI_NO_PATH_UPDATE`| If set the `$PATH` will not be updated to add `pixi` to it.                        |                       |
    | `TMP_DIR`            | The temporary directory the script uses to download to and unpack the binary from. | `/tmp`                |

    For example, on Apple Silicon, you can force the installation of the x86 version:
    ```shell
    PIXI_ARCH=x86_64 curl -fsSL https://pixi.sh/install.sh | bash
    ```
    Or set the version
    ```shell
    PIXI_VERSION=v0.18.0 curl -fsSL https://pixi.sh/install.sh | bash
    ```

=== "Windows"

    The installation script has several options that can be manipulated through environment variables.

    | Variable         | Environment variable | Description                                                                       | Default Value               |
    |------------------|----------------------|-----------------------------------------------------------------------------------|-----------------------------|

>     | `PixiVersion`    | `PIXI_VERSION`       |The version of pixi getting installed, can be used to up- or down-grade.         | `latest`                      |

Author: ```suggestion
    | `PixiVersion`    | `PIXI_VERSION`       |The version of pixi getting installed, can be used to up- or down-grade.           | `latest`                      |
```

>     | `PixiVersion`    | `PIXI_VERSION`       |The version of pixi getting installed, can be used to up- or down-grade.           | `latest`                      |

Author: ```suggestion
    | `PixiVersion`    | `PIXI_VERSION`       |The version of pixi getting installed, can be used to up- or down-grade.           | `latest`                    |
```

>     | `PixiVersion`    | `PIXI_VERSION`       |The version of pixi getting installed, can be used to up- or down-grade.         | `latest`                    |
>     | `PixiHome`       | `PIXI_HOME`          | The location of the installation.                                                 | `$Env:USERPROFILE\.pixi`   |

Reviewer: ```suggestion
    | `PixiVersion`    | `PIXI_VERSION`       |The version of pixi getting installed, can be used to up- or down-grade.         | `latest`                      |
    | `PixiHome`       | `PIXI_HOME`          | The location of the installation.                                                 | `$Env:USERPROFILE\.pixi`    |
```
